windows-gaming-driver/src/setup/vfio.rs

```diff
@@ -79,6 +86,93 @@ pub fn setup(setup: &mut SetupConfig, machine: &mut MachineConfig) -> bool {
     true
 }
 
+fn select(setup: &mut SetupConfig, machine: &mut MachineConfig) -> Result<()> {
+    let udev = Context::new().expect("Failed to create udev context");
+    let mut iter = Enumerator::new(&udev)?;
+    iter.match_subsystem("pci")?;
+    let pci_devs: Vec<_> = iter.scan_devices()?.map(PciDevice::new).collect();
+
+    
+	// filter to the display controller class (0x03XXXX, udev drops the leading zero)
+	let (gpu_devices, gpu_dev_ids, pci_devs) = ask_selection("Which graphics card do you want to pass through?", &|x| x.pci_class.starts_with("3") && x.pci_class.len() == 5, pci_devs, true);
+	
+	println!("Add aditional  pci devices(1/2)");
+	println!();
+	println!("Please choose aditional pci devices. These will be passed throught to qemu on boot.");
+	println!("NOTE: The devices listed here can be reset. They will be bound to the vfio-pci driver on qemu's start and unbound when it quits");
+	println!("This should usually work. If it doesn't ur on ur own.");
+	println!("You will not be able to use these devices while qemu is running.");
+	println!("If the device you want to pass through is not resettable it will show up in the next step.");
+	println!("If you want a device permanently bound to vfio-pci for some reason select it in the next step.");
+	println!("MAKE SURE YOU DON'T PASS THROUGH YOUR USB-CONTROLLER TO WHICH YOUR KEYBOARD AND MOUSE IS CONNECTED!");
+	println!("Helpful tools to avoid this and figure out which numbers are what devices are lspci and lsusb with the -v, -t (lsusb only) and -nn (lspci only) command.");
+	println!();
+	let (temp_devices, _, pci_devs) = ask_selection("Which PCI Devices do you want to pass through temporarily?", &|x| Path::new(PCI_DEVS_PATH).join(x.pci_slot.clone()).join("reset").exists(), pci_devs, false);
```

Comment:
FYI udev devices have `syspath()`

this is very definitely the best way to access this path (check all the other udev code, that's why we don't have any existing "/sys/bus/pci" constants)

Now this *might* be hidden because we convert PCI devices to our own `PciDevice` struct but the fix in that case is to simply change `PciDevice` to carry the syspath.